src/libs/ck-libs/NDMeshStreamer/NDMeshStreamer.h

```diff
@@ -1016,9 +1066,14 @@ class ArrayMeshStreamer :
                 " are not guaranteed to be correct is currently"
                 " not supported.");
       }
-      dtype dataItem;
-      PUP::fromMemBuf(dataItem,(void*)data,size);
-      misdeliveredItems[arrayId].push_back(dataItem);
+      if (!is_fixed_size<dtype>::value) {
+        dtype dataItem;
+        PUP::fromMemBuf(dataItem,(void*)data,size);
+        misdeliveredItems[arrayId].push_back(dataItem);
+      }
+      else {
+        misdeliveredItems[arrayId].push_back(const_cast<dtype&>(*reinterpret_cast<const dtype*>(data)));
```

Comment:
This assumes that anything that is fixed_size is also PUPbytes or trivially serialized. I think fixed_size is either a misnomer or this needs to be changed. In general, we want to optimize TRAM for PUPbytes datatypes so that we can avoid the overheads of PUPing whenever possible, but I'm not sure if we've decided on changing the semantics for that or not.

Reply:
Yes, fixed_size here would actually refer to PUPbytes types. I'll rename it.